src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
+// Possible workaround by adding passport/driving license, but it will cause every room detail check to require to need to enter passport which is inefficient.
+/**
+ * Get room details by entering name of guest staying in that room
+ * @param guestName This is the name of the guest.
+ */
+public void GetRoomDetailByGuestName(String guestName) {
+	Room room = SearchList(guestName);
+	system.out.println("Room with same guest name found");
+	room.viewRoomDetails;
 
-	public void SearchList(String search_text) {
-		// TODO - implement RoomManager.SearchList
-		throw new UnsupportedOperationException();
-	}
+/**
+ * Checks whether a room is vacant or not.(currently)
+ * @param roomNumber Room number of room to be checked.
+ */
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
+		for(var i=0; i<room_list_.size();i++) {
+			if (room_list_.get(i).getRoomNumber()==roomNumber) {
+				room_list_.get(i).checkRoomAvailability();;
+				break;
+			}
+		}
+}		
 
-	public List<Room> GetList() {
-		// TODO - implement RoomManager.GetList
-		throw new UnsupportedOperationException();
-	}
 
-	public void InitializeDB() {
-		// TODO - implement RoomManager.InitializeDB
-		throw new UnsupportedOperationException();
-	}
+public void InitializeDB() {
+	// TODO - implement RoomManager.InitializeDB
+	throw new UnsupportedOperationException();
+}
 
-	public void SaveDB() {
-		// TODO - implement RoomManager.SaveDB
-		throw new UnsupportedOperationException();
-	}
+public void SaveDB() {
+	// TODO - implement RoomManager.SaveDB
+	throw new UnsupportedOperationException();
+}
 
-	/**
-	 * 
-	 * @param guest
-	 * @param room
-	 */
-	public void CheckInRoom(Guest guest, Room room) {
-		// TODO - implement RoomManager.CheckInRoom
-		throw new UnsupportedOperationException();
+/** Assign a guest to a room, guest is stored as an element in a guest list
+ * 
+ * @param guest This is the guest to be added
+ * @param roomNumber This is the room number of the room that the guest will be added to
+ */
+public void CheckInRoom(Guest guest, String roomNumber) {//consider printing out room is occupied if user atttempts to add guest to room awlr with ppl
+	for(var i=0; i<room_list_.size();i++) {
+		if (room_list_.get(i).getRoomNumber()==roomNumber) {
+			room_list_.get(i).addGuest(guest);
+			room_list_.get(i).setStatus(Occupied);
+			break;
+		}
+		else
+			System.out.println("room number does not exist");
 	}
-
-	/**
-	 * 
-	 * @param room
-	 */
-	public void CheckOutRoom(Room room) {
-		// TODO - implement RoomManager.CheckOutRoom
-		throw new UnsupportedOperationException();
+	throw new UnsupportedOperationException();
+}
+/**
+ * removes all guests elements from the guest list of a room
+ * @param roomNumber This is the room number to remove guest elements from.
+ */
+public void CheckOutAllGuest(String roomNumber) {
```

Comment:
Suggest to combine CheckOutAllGuest and CheckOutOneGuest to CheckOutRoom(int room_number). You only need to clear the room's guest list and update room status.

Reply:
This requires discussion